polling/nf_configuration_test.go

```diff
@@ -0,0 +1,261 @@
+
+			if !reflect.DeepEqual(poller.currentPlmnConfig, tc.newPlmnConfig) {
+				t.Errorf("Expected PLMN config to be updated to %v, got %v", tc.newPlmnConfig, poller.currentPlmnConfig)
+			}
+			select {
+			case receivedPlmnConfig := <-ch:
+				if !reflect.DeepEqual(receivedPlmnConfig, tc.newPlmnConfig) {
+					t.Errorf("Expected config %v, got %v", tc.newPlmnConfig, receivedPlmnConfig)
+				}
+			case <-time.After(100 * time.Millisecond):
+				t.Errorf("Expected config to be sent to channel, but it was not")
+			}
+		})
+	}
+}
+
+func TestHandlePolledPlmnConfig_ConfigDidNotChanged_ConfigIsNotSendToChannel(t *testing.T) {
+	testCases := []struct {
+		name          string
+		newPlmnConfig []models.PlmnId
+	}{
+		{
+			name:          "Same config",
+			newPlmnConfig: []models.PlmnId{{Mcc: "001", Mnc: "02"}},
+		},
+		{
+			name:          "Empty config",
+			newPlmnConfig: []models.PlmnId{},
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			ch := make(chan []models.PlmnId, 1)
+			poller := nfConfigPoller{
+				currentPlmnConfig: tc.newPlmnConfig,
+				plmnConfigChan:    ch,
+			}
+			poller.handlePolledPlmnConfig(tc.newPlmnConfig)
+
+			if !reflect.DeepEqual(poller.currentPlmnConfig, tc.newPlmnConfig) {
+				t.Errorf("Expected PLMN list to remain unchanged, got %v", poller.currentPlmnConfig)
+			}
+
+			select {
+			case receivedPlmnConfig := <-ch:
+				t.Errorf("Config was not expected, got %v", receivedPlmnConfig)
+			case <-time.After(100 * time.Millisecond):
+				// Expected case
+			}
+		})
+	}
+}
+
+func TestFetchPlmnConfig(t *testing.T) {
+	validPlmnList := []models.PlmnId{
+		{Mcc: "001", Mnc: "01"},
+		{Mcc: "002", Mnc: "02"},
+	}
+	validJson, err := json.Marshal(validPlmnList)
+	if err != nil {
+		t.Fail()
+	}
+
+	tests := []struct {
+		name           string
+		statusCode     int
+		contentType    string
+		responseBody   string
+		expectedError  string
+		expectedResult []models.PlmnId
+	}{
+		{
+			name:           "200 OK with valid JSON",
+			statusCode:     http.StatusOK,
+			contentType:    "application/json",
+			responseBody:   string(validJson),
+			expectedError:  "",
+			expectedResult: validPlmnList,
+		},
+		{
+			name:          "200 OK with invalid Content-Type",
+			statusCode:    http.StatusOK,
+			contentType:   "text/plain",
+			responseBody:  string(validJson),
+			expectedError: "unexpected Content-Type: got text/plain, want application/json",
+		},
+		{
+			name:          "400 Bad Request",
+			statusCode:    http.StatusBadRequest,
+			contentType:   "application/json",
+			responseBody:  "",
+			expectedError: "server returned 400 error code",
+		},
+		{
+			name:          "500 Internal Server Error",
+			statusCode:    http.StatusInternalServerError,
+			contentType:   "application/json",
+			responseBody:  "",
+			expectedError: "server returned 500 error code",
+		},
+		{
+			name:          "Unexpected Status Code 418",
+			statusCode:    418,
```

Comment:
Why not the following?

```suggestion
			statusCode:    http.StatusTeapot,
```